File: DotNet/SETPractice/DesignPattern/BuilderPattern/PetrolCar.cs
namespace DesignPattern.BuilderPattern
{
    public class PetrolCar
    {
        public ICarBuilder CarBuilder { get; set; }

        public PetrolCar(ICarBuilder carBuilder) 
        { 
            CarBuilder = carBuilder;
        }

        public Car AssembleCar()
        {
            CarBuilder.SetBody();
            CarBuilder.SetSeats(5);
            CarBuilder.SetHeadLights(4);
            CarBuilder.SetPetrolEngine();
            return CarBuilder.GetCar();
        }

    }
}

File: Java/SETPractice/src/main/java/DesignPattern/BuilderPattern/PetrolCar.java
package DesignPattern.BuilderPattern;

public class PetrolCar {
    public ICarBuilder carBuilder;

    public PetrolCar(ICarBuilder carBuilder) {
        this.carBuilder = carBuilder;
    }

    public Car AssembleCar() {
        carBuilder.SetBody();
        carBuilder.SetDieselEngine();
        carBuilder.SetHeadLights(4);
        carBuilder.SetSeats(5);
        return carBuilder.GetCar();
    }
}
